bench/benches/packets.rs

```diff
@@ -1,22 +1,47 @@
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use criterion::{criterion_group, criterion_main, Criterion};
 use nb2::packets::ip::v4::Ipv4;
 use nb2::packets::{Ethernet, Packet, Udp};
-use nb2::testils::byte_arrays::UDP_PACKET;
+use nb2::testils::criterion::BencherExt;
+use nb2::testils::proptest::*;
+use nb2::testils::PacketExt;
 use nb2::Mbuf;
+use proptest::prelude::*;
 
-fn parse_ipv4_packet() {
-    let packet = Mbuf::from_bytes(&UDP_PACKET).unwrap();
-    let ethernet = packet.parse::<Ethernet>().unwrap();
+const BATCH_SIZE: usize = 500;
+
+fn parse_udp_packet(mbuf: Mbuf) {
+    let ethernet = mbuf.parse::<Ethernet>().unwrap();
     let ipv4 = ethernet.parse::<Ipv4>().unwrap();
     ipv4.parse::<Udp<Ipv4>>().unwrap();
 }
 
-#[nb2::bench]
+fn deparse_udp_packet(udp: Udp<Ipv4>) {
```

Comment:
should return the packets from these functions. if you don't, then the mbufs will be freed inside the fn and cause that to skew the bench result.

Reply:
Ah yep.